### ISSUE TITLE
vApp properties of type password continuously showing diff

### ISSUE THREAD

**Issue by Yousseph** [hashicorp/terraform-provider-vsphere#922]
### Terraform Version
Terraform v0.12.17

### vSphere Provider Version
provider.vsphere v1.13.0

### Affected Resource(s)
- `vsphere_virtual_machine`

### Terraform Configuration Files
```hcl
resource "vsphere_virtual_machine" "mini-staging" {
  name             = "mini-staging"
  folder           = vsphere_folder.vmware-demo-staging.path
  resource_pool_id = data.vsphere_resource_pool.pool.id
  datastore_id     = data.vsphere_datastore.datastore.id

  num_cpus = 2
  memory   = 4096
  guest_id = data.vsphere_virtual_machine.mini-staging-template.guest_id
  scsi_type = data.vsphere_virtual_machine.mini-staging-template.scsi_type
  firmware = data.vsphere_virtual_machine.mini-staging-template.firmware

  network_interface {
    network_id   = data.vsphere_network.network.id
     adapter_type = data.vsphere_virtual_machine.mini-staging-template.network_interface_types[0]
  }
  
  wait_for_guest_net_routable = true

  lifecycle {
    ignore_changes = [annotation]
  }

  disk {
    label            = "disk0"
    size             = data.vsphere_virtual_machine.mini-staging-template.disks.0.size
    eagerly_scrub    = data.vsphere_virtual_machine.mini-staging-template.disks.0.eagerly_scrub
    thin_provisioned = data.vsphere_virtual_machine.mini-staging-template.disks.0.thin_provisioned
  }

  clone {
      template_uuid = data.vsphere_virtual_machine.mini-staging-template.id
  }

  vapp {
    properties = {
      "USER_NAME" = "jhonathan"
      "PASSWORD" = "mynewpass"
    }
  }
}
```

### Debug Output
![image](https://user-images.githubusercontent.com/22526418/70157381-c6e4da80-16b5-11ea-9bb9-f53fc2b42f0b.png)
![image](https://user-images.githubusercontent.com/22526418/70156731-b08a4f00-16b4-11ea-9908-72db6f1b3339.png)

### Expected Behavior
There should be no diff in subsequent terraform plan runs: no changes made in terraform manifests nor in the VM side

### Actual Behavior
Terraform plan reports diff in the vapp.property section: 1 property added called **PASSWORD**

### What actually happened?
The vapp property called **PASSWORD** has never been changed so it should not generate a diff in the terraform plan
Note that: there no issue with the **USER_NAME** vapp property, the issue occurs only with the vapp properties with type **password**
![image](https://user-images.githubusercontent.com/22526418/70157988-c731a580-16b6-11ea-8730-0505a6377fad.png)

### Steps to Reproduce
1. create a vmware template with vapp enabled
2. create a property with type **password** (not string)
3. in the terraform manifest clone a VM from the previous template and customize it by setting value to the vapp property
4. run terraform plan and terraform apply to create the VM
5. after VM is created run terraform plan without any change
6. check that terraform plan report a diff related to the vapp property section

**Comment by hashibot** [comment]
This issue has been open 180 days with no activity. If this issue is reproducible with the latest version of the provider and with Terraform 0.12, please comment. Otherwise this issue will be closed in 30 days.

**Comment by Yousseph** [comment]
any update of this?

**Comment by bill-rich** [comment]
Hi @Yousseph! Is it possible that the password being supplied in the configuration is the same as the default password in the OVF? If that is the case, this is expected behavior to allow users to leave the OVF defaults by not supplying them in the configuration.

**Comment by Yousseph** [comment]
Hi @bill-rich, No, it's not the default, in the case described above the default is an empty string for all vApp properties
This is not happening with regular vApp properties of type "string": if there is no change the terraform plan doesn't show any diff
For this reason, I'm not using the "password" type for secrets: to overcome this I have to encrypt the value and use the "string" type, fixing this issue will unblock us and we can finally get rid of the encrypt/decrypt overhead

**Comment by heitmanr** [comment]
same here: VM get's shutdown/modified/poweredon everytime,

logging
```HCL
~ vapp {
          ~ properties = {
              + "guestinfo.password"  = "VMware1!3"
              + "guestinfo.ssh"       = "True"
                # (7 unchanged elements hidden)
            }
        }
```

everything set after first run
![grafik](https://user-images.githubusercontent.com/26636908/151075646-f9f301cc-2d08-4e40-a462-6a59cc93948f.png)

as configured statically
```HCL
resource vsphere_virtual_machine "vesx" {
...
vapp {
    properties = {
      ...
      "guestinfo.password" = "VMware1!3",
      "guestinfo.ssh" = "True"
    }
}
```
The password here is similar to the default-password "VMware1!" but different ;-)

**Comment by heitmanr** [comment]
thinking about this for ten additional minutes - for my setup the vapp-properties are only relevant for the very first start of the virtual-machine.

So adding
```HCL
resource vsphere_virtual_machine "vesx" {
vapp {
...
}
lifecycle {
   ignore_changes = [
     vapp[0].properties,
   ]
}

}
```
fixes the issue - now terraform ignores the by mistake detected non-existing diff regarding the password..

**Comment by tenthirtyam** [comment]
This! ☝🏻

**Comment by tenthirtyam** [comment]
Also, if there are **specific** properties would wish to ignore you can drill down to the property item.

resource vsphere_virtual_machine "foo" {
...
 vapp {
 ...
 }
 lifecycle {
   ignore_changes = [
     vapp[0].properties.username,
     vapp[0].properties.password,
   ]
 }

}

**Comment by tenthirtyam** [comment]
@Yousseph, although your issue is rather old, could you review the prior comments and confirm if this resolves your concerns.

**Comment by tenthirtyam** [comment]
Marking as closed based on the previous comments and solutions provided in the thread ([1](https://github.com/hashicorp/terraform-provider-vsphere/issues/922#issuecomment-1021721476)) ([2](https://github.com/hashicorp/terraform-provider-vsphere/issues/922#issuecomment-1021725156)).

If the issue persists with the solution provided and [the latest version of the provider](https://github.com/hashicorp/terraform-provider-vsphere/releases/latest), please open a new issue and reference this issue for additional context.

**Ryan Johnson**
Staff II Solutions Architect at VMware, Inc.
Cloud Infrastructure Business Group, Solutions Engineering
